Week 2 Assignments/Linear Regression with Multiple Variables/mlclass-ex1/featureNormalize.m

```diff
@@ -13,27 +13,30 @@
 % ====================== YOUR CODE HERE ======================
 % Instructions: First, for each feature dimension, compute the mean
 %               of the feature and subtract it from the dataset,
-%               storing the mean value in mu. Next, compute the 
+%               storing the mean value in mu. Next, compute the
 %               standard deviation of each feature and divide
 %               each feature by it's standard deviation, storing
-%               the standard deviation in sigma. 
+%               the standard deviation in sigma.
 %
-%               Note that X is a matrix where each column is a 
-%               feature and each row is an example. You need 
-%               to perform the normalization separately for 
-%               each feature. 
+%               Note that X is a matrix where each column is a
+%               feature and each row is an example. You need
+%               to perform the normalization separately for
+%               each feature.
 %
 % Hint: You might find the 'mean' and 'std' functions useful.
-%       
+%
+
 
+mu = mean(X);
+sigma = std(X);
 
-mu = mean(X_norm);
-sigma = std(X_norm);
+% Note: It is not really necessary to repmat the mu and sigma matrix
+% but if you want you can repmat those two
+% mu = repmat(mu, size(mu, 2), 1);
+% sigma = repmat(sigma, size(sigma, 2), 1);
 
-tf_mu = X_norm - repmat(mu,length(X_norm),1);
-tf_std = repmat(sigma,length(X_norm),1);
 
-X_norm = tf_mu ./ tf_std;
+X_norm = (X - mu) ./ sigma;
```

Comment:
Not sure if this works for Matlab, but in Octave, this doesn't work.

Moreover, it seems like this only works on version Matlab version 2016 or later. 

`In MATLAB® R2016b and later, you can directly use operators instead of bsxfun, since the operators independently support implicit expansion of arrays with compatible sizes.`

https://www.mathworks.com/help/matlab/ref/bsxfun.html

Since Coursera targets both Octave and Matlab users, there should be an annotation in your code that this statement works ONLY for some versions of Matlab.